docs/architecture/decisions/0010-release-and-deployment-process-for-azure-functions.md

```diff
@@ -0,0 +1,70 @@
+# 10. Release and deployment process for Azure Functions
+
+Date: 2017-10-20
+
+## Status
+
+Accepted
+
+## Context
+
+The Digital Citizenship APIs are based on the Azure Functions framework (a _serverless_ application framework). 
+
+During the initial development of the APIs, we relied on the [continuous deployment feature](https://docs.microsoft.com/en-us/azure/azure-functions/functions-continuous-deployment), configured to deploy the API code from the `master` branch of the [digital-citizenship-functions](https://github.com/teamdigitale/digital-citizenship-functions) repository.
+
+This release process served us well during the initial development phase, but it has several drawbacks:
+
+1. there's no approval process that triggers the deployment of the `master` branch into production
+1. the build phase happens on the production hosts
+1. we cannot control the actual deployment process (the build and the re-load of the code happens automatically on the production hosts)
+1. Rolling back is cumbersome as it requires revering commits on the `master` branch.
+
+## Decision
+
+We decide to structure the release and deployment process in the following steps:
+
+### 1 - Automated tests
+
+On commit to the `master` branch, the CI system checks out the commit and runs the automated tests.
+
+### 2 - Build of release artifacts
+
+On tests passing successfully, the CI workflow triggers the build process that generates artifacts needed for release (e.g. packs dependencies, compiles Typescript into Javascript, etc...).
```

Comment:
we cite a  'CI workflow' but it looks like it's not described anywhere

Reply:
yes, meaning we need to create on, I'll rephrase